mixins/scrollBetweenSections.ts

```diff
@@ -0,0 +1,62 @@
+import Vue from 'vue'
```

Comment:
Non-blocking: the trivial factorization is OK. However, making this really reusable would imply separating the concerns of detecting-active-section/scrolling (there is another factorization opportunity here considering the behaviour of the learn page), invoke callbacks (for instance, upon parsing the URL, instead of always changing the section, you invoke `onParseUrl`; or upon a section being marked as active, you invoke `onActiveSectionChange`) so you decouple the detection behaviour from its consequences.

Instead of a mixin, the "scroll with active section detection" could be a component admitting parameters for customizing the detection window and emitting events when some things happen.

Ultimately, there is room for improvement so I would open a follow-up (you can use this comment as a starting point) or retire the factorization and keep track of it in an issue.